### PR TITLE
Polyfill string.matchAll to unblock #7770

### PR DESCRIPTION
**Pull request by mattsoulanille**
Manually write a polyfill for string.matchAll to unblock #7770. This should be replaced with core-js in the future.

To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

### CODE REVIEW

**Comment by mattsoulanille** [comment]
@pforderique This should unblock your PR

**Review by fengwuyao — APPROVED** [review]
https://pantheon.corp.google.com/cloud-build/builds/7de8aadf-cfa2-4088-a605-df84c86c3341;step=4?project=learnjs-174218

Some lint.

**Review by pforderique — APPROVED** [review]
LGTM!

File: tfjs-layers/src/layers/nlp/match_all_polyfill.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+// TODO(mattSoulanille): Replace this with automatic polyfilling using core-js.
+export function *matchAll(str: string, regexp: RegExp): IterableIterator<RegExpMatchArray> {
+  // Remove the global flag since str.match does not work with it.
+  const flags = new Set(regexp.flags.split(''));
```
Comment (chunnienc) [inline tfjs-layers/src/layers/nlp/match_all_polyfill.ts:21]: nit: `flags.replace(/g/g, '')`

**Review by chunnienc — APPROVED** [review]

**Pull request merged**